selfdrive/pandad.py

```diff
@@ -34,6 +37,24 @@ def flash_panda(panda_serial: str) -> Panda:
 
   if panda.bootstub or panda_signature != fw_signature:
     cloudlog.info("Panda firmware out of date, update required")
+
+    if TICI and not panda.bootstub:
+      panda.reset(enter_bootstub=True)
+      if not panda.bootstub:
+        cloudlog.error(f"Panda unable to enter bootstub. Attempting to recover.")
+
+        gpio_set(GPIO.STM_RST_N, 1)
+        gpio_set(GPIO.STM_BOOT0, 1)
+        time.sleep(2)
+        gpio_set(GPIO.STM_RST_N, 0)
+
+        panda.recover(skip_enter_dfu=True)
```

Comment:
Should be fine to release `BOOT0` a second or so after releasing reset, that way you don't have to reset again after the recovery.

Also, `STM_RST_N` would suggest that it would reset when set to 0, which isn't the case. Would rename that to `STM_RST` to avoid confusion :)